tests/test_search.py

```diff
@@ -0,0 +1,24 @@
+from io import BytesIO
+
+from lxml import etree
+
+
+def test_empty_response(client) -> None:
+    response = client.get('/search')
+    xml = etree.parse(BytesIO(response.data))
+    root = xml.getroot()
+
+    assert root.getchildren() == []
+
+
+def test_empty_response_is_well_formed(client) -> None:
+    response = client.get('/search')
+    xml = etree.parse(BytesIO(response.data))
+    root = xml.getroot()
+
+    assert response.status_code == 200
+    assert response.content_type == 'application/xml; charset=utf-8'
+    assert xml.docinfo.encoding == 'UTF-8'
+    assert xml.docinfo.xml_version == '1.0'
+    assert xml.docinfo.root_name == 'item-list'
+    assert root.tag == '{http://libero.pub}item-list'  # check namespace
```

Comment:
This all looks rather unsustainable: not that many lines for an empty list, but as soon as there's one with content...

In fact, doesn't actually check that the list in empty?

Still think an [equivalent of `assertXmlStringEqualsXmlString`](https://github.com/libero/content-api-bundle/blob/5489da6667e68f7f4d2f5624de1abcb6846d4800/tests/Functional/ItemsTest.php#L37-L40) is a better approach.

Reply:
> In fact, doesn't actually check that the list in empty?

I suggested a separate test for that case, since this one is oriented to "schema" checks more than content. I do think checking actual listings will be different and this is a test not to be replicated: in the case of a listing we would extract the list of elements with Xpath and check the presence of what we have added but you don't want to write
```
assert response == \
           '<item-list xmlns="http://libero.pub">' \
           '<item-ref id="42">' \
           '</item-list>'
```
when you can write:
```
    self.assertFoundItems([42])
```
or even
```
    self.assertFoundItems([42])
    self.assertNotFoundItems([43])
```
for negative cases and similar.

Reply:
Full expected responses have their place to demonstrate a sample of what it should look like, but this is a service which returns a flat list so there's not much explanatory power in replicating the full document for all cases.

Even with JSON you would never write:
```
    assertJsonEquals(items, '[{"key":"value", ...}]')
```
but the expected value would still be a data structure in the source code, as a string is prone to syntax errors.

Reply:
@thewilkybarkid @giorgiosironi I can't find anything like the php assertion method in python. Also, it seems it's not trivial to compare python xml tree objects. `lxml` has the ability to check a document against a schema. Would this be a better way to test xml?